munge/train-test.R

```diff
@@ -0,0 +1,75 @@
+full_bbs = na.omit(readRDS("data/munged.Rds"))
+library(fastICA)
+library(magrittr)
+library(dplyr)
+library(beepr)
+
+focal_year = 1990
+
+set.seed(1)
+
+# Train on the training data through focal_year
+train_bbs = full_bbs[full_bbs$in_train & full_bbs$year <= focal_year, ]
+
+make_rescaler = function(x){
+  scaled_x = scale(x)
+  
+  # Half the variance in X should come from temp and half from precip,
+  # but 3/4 of the columns are about temperature.
+  temp_columns = grep("^t", colnames(scaled_x))
+  scaled_x[ , temp_columns] = scaled_x[ , temp_columns] / sqrt(3)
+  
+  ica = fastICA(scaled_x, 10)
+  
+  function(new_x){
```

Comment:
This must be something I don't understand about R yet. How does this nested function end up getting executed?